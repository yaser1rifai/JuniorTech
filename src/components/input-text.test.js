import React from 'react';
import InputText  from '../App';

import Enzyme,  {shallow,mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


describe('Input Text component',()=>{

let component;
let props;
let label= 'keywords';
let keywords='this is some keywords'

Enzyme.configure({
 adapter: new Adapter()
});
//------------------------
beforeEach(()=>{
    props={
        onChange: jest.fn()
    }
     component=shallow(
   <InputText 
   label={label}
   value={keywords} 
   onChange={props.onChange} />
);
})
//-------------------------

    it('should render the right way ',()=>{
        //expect(true).toBe(true)
        expect(component).toMatchSnapshot();
    });

    /* it('should render the label  the right way ',()=>{
        //expect(true).toBe(true)
        expect(component).find('label').text().toBe(label+':');
    }); */

   /*  it('should call the onChange function ',()=>{
        let mountedComponent = mount(
      <InputText 
     label={label}
     value={keywords} 
     onChange={props.onChange} />
        );
        mountedComponent.find('input').simulate('change') ;
        expect(props.onChange).toHaveBeenCalled();
      }); */

})