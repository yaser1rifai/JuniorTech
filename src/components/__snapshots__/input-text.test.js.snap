// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Text component shoud render the right way  1`] = `ShallowWrapper {}`;

exports[`Input Text component should render the right way  1`] = `ShallowWrapper {}`;
